src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisScope.cs

```diff
@@ -117,6 +134,9 @@ public AnalysisScope WithAnalyzers(ImmutableArray<DiagnosticAnalyzer> analyzers,
             return new AnalysisScope(SyntaxTrees, AdditionalFiles, analyzers, isPartialAnalysis, FilterFileOpt, FilterSpanOpt, IsSyntacticSingleFileAnalysis, ConcurrentAnalysis, CategorizeDiagnostics);
         }
 
+        public AnalysisScope WithFilterSpan(TextSpan? filterSpan)
+            => new(SyntaxTrees, AdditionalFiles, Analyzers, IsPartialAnalysis, FilterFileOpt, filterSpan, IsSyntacticSingleFileAnalysis, ConcurrentAnalysis, CategorizeDiagnostics);
```

Comment:
>new [](http://example.com/codeflow?start=15&length=3)

I can accept that ```DiagnosticAnalyzer``` is "grey area" in terms of what style rules to follow. However, generally in compiler we ban target typed new.